project/categories/routes.py

```diff
@@ -0,0 +1,156 @@
+# Modules from Flask Library
+from flask import Flask, render_template, request, flash, Blueprint
+from flask import session as login_session
+from flask import make_response, redirect, jsonify, url_for
+# Access database
+from project.models import Category, Product, User
+from project.db import DBConnector
+# Services
+from project.services.categories import CategoryService
+from project.services.auth import AuthService
+from project.services.products import ProductService
+
+import random
+import string
+import httplib2
+import json
+import requests
+
+
+# Create Blueprint 'categories'
+categories = Blueprint('categories', __name__)
+
+# Connect to the database, create session
+session = DBConnector().get_session()
+
+# Instantiate services
+category_service = CategoryService()
+product_service = ProductService()
+auth_service = AuthService()
+
+
+@categories.route('/')
+@categories.route('/categories/')
+def showCategories():
+    """App route function for main page to show all categories."""
+    return render_template('index.html',
+                           categories=category_service.get_all_categories(),
+                           isLogin=auth_service.is_user_authorized(),
+                           latestProducts=(product_service
+                                           .get_latest_products(8)))
+
+
+@categories.route('/categories/<int:cat_id>/')
+def showCategoryProducts(cat_id):
+    """App route function to show Products of the selected Category."""
+    isLogin = auth_service.is_user_authorized()
+    # Pick category selected by user
+    category = category_service.get_category_by_id(cat_id)
+    # Check if user is Creator of Category
+    isCreator = False
+    if isLogin:
+        isCreator = login_session['email'] == category.user.email
+    return render_template('category.html',
+                           products=(product_service
+                                     .get_products_by_cat_id(cat_id)),
```

Comment:
Same with method names:
```suggestion
                                     .get_products_by_category_id(category_id)),
```